Move profilecli docs to a dedicated page, add installation instructions

aleks-p opened the pull request:
Improves profilecli docs as suggested in #2650 

A few remarks (cc. @Rperry2174):

1. I kept the installation instructions simple for now. We can technically use package managers on Linux as well but in my tests they also start a Pyroscope server as well which is not what we want in this context.
2. After looking at the [examples/api folder](https://github.com/grafana/pyroscope/tree/main/examples/api) more carefully, I am not sure examples for a CLI tool belong there. One option could be a bash script, perhaps in a new `examples/cli` directory. This will however be a one-liner that repeats what the docs already say. Any thoughts?

cc. @knylander-grafana

Rperry2174 commented:
> After looking at the [examples/api folder](https://github.com/grafana/pyroscope/tree/main/examples/api) more carefully, I am not sure examples for a CLI tool belong there. One option could be a bash script, perhaps in a new examples/cli directory. This will however be a one-liner that repeats what the docs already say. Any thoughts?

Good point. Let's leave it like this for now.. I do think we should still at least mention and link to this page somewhere on the "server api" page since a lot of people may start on server api page and ultimately what they're really looking for / need is profilecli

kolesnikovae commented:
It would be great to also add instructions for `query` sub-command – that is super useful if one wants to integrate PGO into a CI pipeline

cyriltovena commented on docs/sources/configure-server/profile-cli.md:
```diff
@@ -0,0 +1,127 @@
+- interacting with a running Pyroscope Server to upload profiles manually, run queries, etc.
+- inspecting [Parquet](https://parquet.apache.org/docs/) files.
+
+Hint: Use the `help` command (`profilecli help`) to get a full list of capabilities as well as additional help information.
```
```suggestion
> Hint: Use the `help` command (`profilecli help`) to get a full list of capabilities as well as additional help information.
```

aleks-p commented:
@Rperry2174 @kolesnikovae I added a short section in the server API page with a link to the profile CLI docs
 as well as instructions for the `query` command. Feel free to update any wording as needed.

Also, in my last commit I switched from flags to environment variables for `url`, `username` and `password` because this is how we are using the commands from what I have seen and is usually the recommended way. Let me know if you have any concerns.

cyriltovena commented on docs/sources/configure-server/profile-cli.md:
````diff
@@ -0,0 +1,234 @@
+For example, for Linux with the AMD64 architecture:
+
+```bash
+curl -fL https://github.com/grafana/pyroscope/releases/download/v1.0.0/profilecli_1.0.0_linux_amd64.tar.gz | tar xvz
````
Should we point to 1.1.5 ?

aleks-p replied:
did it for now, though this will get out of date quickly unless we generate these parts of the docs

cyriltovena commented on docs/sources/configure-server/profile-cli.md:
```diff
@@ -0,0 +1,234 @@
+
+   - You can provide a label selector using the `--query` flag (e.g., `--query='{service_name="my_application_name"}'`)
+   - You can provide a custom time range using the `--from` and `--to` flags (e.g., `--from="now-3h" --to="now"`)
+   - You can specify the profile type via the `--profile-type` flag. The available profile types are listed in the output of the `profilecli query series` command.
```
Makes me think we should make this simpler, with flag like `-mem` or `-cpu` to simplify the usage of this.

We should also most likely show using go how to open and read those merged profiles. I suspect most people don't know how to use pprof.

cyriltovena reviewed (approved):
LGTM, left some suggestions for follow up.
